### PR TITLE
Ensure that PythonAutoContainerTask has a container_image attribute

### PR DESCRIPTION
**Pull request by pingsutw**
## Why are the changes needed?

Fix unit test. https://github.com/flyteorg/flytekit/actions/runs/9620004661/job/26621714719

## What changes were proposed in this pull request?
Call `super().__init__` after setting container_image because PythonAutoContainerTask
is added to the FlyteEntities in `super().__init__`. The translator will iterate over
FlyteEntities and call container_image. Therefore, we should ensure the container_image attribute is set
before appending the task to FlyteEntities.

https://github.com/flyteorg/flytekit/blob/876877abd8d6f4f54dec2738a0ca07a12e9115b1/flytekit/tools/translator.py#L181

## How was this patch tested?
unit test

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs
NA

## Docs link
NA

### CODE REVIEW

File: .github/workflows/monodocs_build.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Monodocs Build
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
```
Comment (eapolinario) [inline .github/workflows/monodocs_build.yml:4]: why are we removing this?

File: flytekit/core/python_auto_container.py [hunk 2]
```diff
@@ -87,19 +87,28 @@ def __init__(
...
+        # Therefore, we need to ensure the _container_image attribute is set
+        # before appending the task to FlyteEntities.
+        # https://github.com/flyteorg/flytekit/blob/876877abd8d6f4f54dec2738a0ca07a12e9115b1/flytekit/tools/translator.py#L181
+
         super().__init__(
             task_type=task_type,
             name=name,
             task_config=task_config,
             security_ctx=sec_ctx,
+            environment=environment,
```
Comment (eapolinario) [inline flytekit/core/python_auto_container.py:109]: do we have to worry about `environment` being `None` after this call? (looking at the initialization to `{}` in the existing code).

**Review by eapolinario — COMMENTED** [review]
I agree in principle with this PR, but I'd like to understand a bit more how this started failing all of a sudden (esp. after https://github.com/flyteorg/flytekit/pull/2485).

File: .github/workflows/monodocs_build.yml (hunk 1 above)
Comment (pingsutw) [reply .github/workflows/monodocs_build.yml:4]: It allows the GA workflow to run against all pushed commits without being canceled when a new PR is merged

File: flytekit/core/python_auto_container.py (hunk 2 above)
Comment (pingsutw) [reply flytekit/core/python_auto_container.py:109]: PythonTask (parent class) will set a default value (`{}`) for it.
https://github.com/flyteorg/flytekit/blob/9d94c83b622b53f65055915efba081470f62c1bc/flytekit/core/base_task.py#L490

File: .github/workflows/monodocs_build.yml (hunk 1 above)
Comment (eapolinario) [reply .github/workflows/monodocs_build.yml:4]: Makes sense. We use this pattern in a [bunch of our repos](https://github.com/search?q=org%3Aflyteorg+github.event.pull_request.number+%7C%7C+github.ref+org&type=code), can you also make sure these are changed?

**Pull request merged**